### PR TITLE
line chart: refactor to separate internal types

### PR DESCRIPTION
**Pull request by stephanwlee**
There are few types that are used internal to the charting library and
there are few public ones. Since the public one is a subset of the
internal one, we are turning `types`, which is too generic, into
`internal_types` and selectively re-export types in the `public_types`.

### CODE REVIEW

File: tensorboard/webapp/widgets/line_chart_v2/lib/renderer/BUILD [hunk 1]
```diff
@@ -25,7 +25,7 @@ tf_ts_library(
         "renderer_types.ts",
```
Comment (psybuzz) [inline tensorboard/webapp/widgets/line_chart_v2/lib/renderer/BUILD:25]: Can we visibility restrict this target to lib:subpackages?  `public_types.ts` only re-exports specific symbols from `renderer_types.ts`, suggesting that some of its symbols may be non-public.

File: tensorboard/webapp/widgets/line_chart_v2/lib/public_types.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+export {
+  DataSeries,
+  DataSeriesMetadata,
+  DataSeriesMetadataMap,
+  Dimension,
+  Extent,
+  Point,
+  Rect,
+} from './internal_types';
```
Comment (psybuzz) [inline tensorboard/webapp/widgets/line_chart_v2/lib/public_types.ts:24]: Looking ahead at [1], I see evidence why most of these need to be public, except for `Rect`.  It seems that `Rect` is useful for anything that deals with a `Coordinator`, which line_chart_component and above should not be exposed to.

Should we drop `Rect`?

[1] https://github.com/tensorflow/tensorboard/blob/swl_gpu_11/tensorboard/webapp/widgets/line_chart_v2/line_chart_component.ts

File: tensorboard/webapp/widgets/line_chart_v2/lib/renderer/BUILD (hunk 1 above)
Comment (stephanwlee) [reply tensorboard/webapp/widgets/line_chart_v2/lib/renderer/BUILD:27]: Done.

File: tensorboard/webapp/widgets/line_chart_v2/lib/public_types.ts (hunk 2 above)
Comment (stephanwlee) [reply tensorboard/webapp/widgets/line_chart_v2/lib/public_types.ts:23]: Done

**Review by psybuzz — APPROVED** [review]

**Pull request merged**